sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferClient.java

```diff
@@ -73,39 +76,133 @@ private TransferClient(Context context,
+
+                Log.v(TAG, "upload(): enqueuing UploadWorker: " + transferId);
+                WorkManager.getInstance(context)
+                    .beginUniqueWork(toTransferUniqueWorkName(transferId),
+                        ExistingWorkPolicy.KEEP,
+                        uploadWorkRequest)
+                    .enqueue();
+                idOrErrorChannel.postValue(TransferIdOrError.id(TransferIdOrError.Operation.UPLOAD_DOWNLOAD, transferId));
+            } catch (Exception e) {
+                idOrErrorChannel.postValue(TransferIdOrError.error(TransferIdOrError.Operation.UPLOAD_DOWNLOAD, e));
+            }
+        });
+        // UI_Thread
+        return toCachedTransferInfoLiveData(idOrErrorChannel, false);
+    }
+
+    /**
+     * Pause a transfer identified by the given transfer id. The pause operation
+     * is a best-effort and it is possible that a transfer that is already executing
+     * may continue to transfer.
+     *
+     * Upon successful scheduling of the pause, any observer observing on
+     * {@link LiveData<TransferInfo>} for this transfer receives a {@link TransferInfo}
+     * event with state {@link TransferInfo.State#USER_PAUSED}.
+     *
+     * @param transferId the transfer id identifies the transfer to pause.
+     */
+    // P2: Currently no return value, evaluate any possible return value later.
+    public void pause(long transferId) {
+        // UI_Thread
+        final TransferIdInfoLiveData.TransferFlags transferFlags = TRANSFER_ID_INFO_CACHE.getTransferFlags(transferId);
+        this.serialTaskExecutor.execute(() -> {
+            // BG_Thread
+            try {
+                final PauseCheck pauseCheck = checkPauseable(transferId);
+                if (pauseCheck.canPause) {
+                    if (pauseCheck.isUpload) {
+                        db.uploadDao().updateUploadInterruptState(transferId, UploadInterruptState.USER_PAUSED);
+                    } else {
+                        throw new RuntimeException("Download::pause() NotImplemented");
+                    }
+                    if (transferFlags != null) {
+                        transferFlags.setUserPaused();
+                    }
+                    WorkManager
+                        .getInstance(context)
+                        .cancelUniqueWork(toTransferUniqueWorkName(transferId));
+                }
+            } catch (Exception e) {
+                Log.e(TAG, "Unable to schedule pause for the transfer:" + transferId, e);
+            }
+        });
+    }
+
+    /**
+     * Resume a transfer that was paused.
+     *
+     * @param @param transferId the transfer id identifies the transfer to resume.
+     * @return LiveData that streams {@link TransferInfo} describing current state of the transfer
+     */
+    public LiveData<TransferInfo> resume(long transferId) {
+        // UI_Thread
+        final MutableLiveData<TransferIdOrError> idOrErrorChannel = new MutableLiveData<>();
```

Comment:
Is it a convention to name MutableLiveData objects channels?

Reply:
there is no convention like that, just my own convention, memories of Go :) We can remove that suffix.

Reply:
I like the `channel` nomenclature myself. It's easier for me to think of `LiveDatas` as just channels.